src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -30,24 +36,66 @@ function HeatmapVis(props: Props): JSX.Element {
     keepAspectRatio = true,
     showGrid = false,
     showLoader = true,
+    abscissas,
+    ordinates,
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
+  const abscissaValues = abscissas || range(cols + 1);
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    return getDataDomain(abscissaValues) ?? DEFAULT_DOMAIN;
+  }, [abscissaValues]);
+
+  const ordinateValues = ordinates || range(rows + 1);
+  const ordinateToIndex = getValueToIndexScale(ordinateValues);
+  const ordinateDomain = useMemo(() => {
+    return getDataDomain(ordinateValues) ?? DEFAULT_DOMAIN;
+  }, [ordinateValues]);
+
+  if (abscissaValues.length !== cols + 1) {
+    return (
+      <p>{`Abscissas size (${
+        abscissaValues.length
+      }) does not match data length along X (${cols + 1})`}</p>
+    );
+  }
+
+  if (ordinateValues.length !== rows + 1) {
+    return (
+      <p>{`Ordinate size (${
+        ordinateValues.length
+      }) does not match data length along Y (${rows + 1})`}</p>
+    );
+  }
```

Comment:
Just another thought: would it be better to throw an error? This is basically an assertion to make sure things don't break later on, so to me, we'd be better off breaking hard.... especially since I'm currently adding an error boundary to handle errors in visualization components! If we don't throw, then the component tree has no way of knowing that an error ocurred.

Reply:
Sure !